Signal/src/ViewControllers/AppSettings/AddToBlockListViewController.m

```diff
@@ -78,6 +80,12 @@ - (void)signalAccountWasSelected:(SignalAccount *)signalAccount
         OWSFail(@"%@ Cannot add already blocked user to block list.", self.logTag);
         return;
     }
+
+    if ([signalAccount.recipientId isEqual:TSAccountManager.localNumber]) {
+        [self showCannotBlockSelfAlert];
+        return;
+    }
+
     [BlockListUIUtils showBlockSignalAccountActionSheet:signalAccount
```

Comment:
Weird. It looks like `BlockListUIUtils showBlockSignalAccountActionSheet` was already checking for this.  Was that check not working?